app/src/test/java/com/codesoom/assignment/controllers/TaskControllerTest.java

```diff
@@ -0,0 +1,53 @@
+package com.codesoom.assignment.controllers;
+
+import com.codesoom.assignment.TaskNotFoundException;
+import com.codesoom.assignment.application.TaskService;
+import com.codesoom.assignment.models.Task;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+class TaskControllerTest {
+    private static final String TASK_TITLE = "test";
+
+    private TaskController controller;
+
+    @BeforeEach
+    void setUp() {
+        TaskService taskService = new TaskService();
+        controller = new TaskController(taskService);
+    }
+
+    @Test
+    void list() {
+        assertThat(controller.list()).isEmpty();
+    }
+
+    @Test
+    void createNewTask() {
+        Task task = new Task();
+        task.setTitle(TASK_TITLE);
+
+        controller.create(task);
+
+        assertThat(controller.list()).isNotEmpty();
+    }
+
+    @Test
+    void getTaskWithValidId() {
+        Task task = new Task();
+        task.setTitle(TASK_TITLE);
+
+        controller.create(task);
+
+        assertThat(controller.detail(1L).getTitle()).isEqualTo(TASK_TITLE);
+    }
+
+    @Test
+    void getTaskWithInvalidId() {
+        assertThatThrownBy(() -> controller.detail(100L))
+                .isInstanceOf(TaskNotFoundException.class);
+    }
```

Comment:
junit5 의 assertThrows를 사용해보면 어떨까요? 다음은 제가 작성해본 것입니다. 이렇게 하면 어떤 장점이 있을까요?
```suggestion
    @Test
    void detailWithInvalidId() {
        Assertions.assertThrows(
            TaskNotFoundException.class,
            () -> taskController.detail(100L),
            "할 일을 찾을 수 없을 경우를 표현하는 예외가 던져져야 합니다"
        );
    }
```

Reply:
함수 내부를 살펴보았는데,
예외가 발생했지만 기대하던 예외가 아닌 경우 / 예외가 발생하지 않은 경우를 구분해서 처리를 해주니
테스트로 더 많은 정보를 쉽게 얻을 수 있을 것 같습니다.

Reply:
함수 내부를 살펴보셨다니 훌륭해요. 칭찬해드리고 싶습니다.